upup/pkg/fi/cloudup/new_cluster.go

```diff
@@ -1408,6 +1415,17 @@ func initializeOpenstackAPI(opt *NewClusterOptions, cluster *api.Cluster) {
 			cluster.Spec.CloudProvider.Openstack.Loadbalancer.FloatingSubnet = fi.PtrTo(opt.OpenstackLBSubnet)
 		}
 	}
+
+	// this is needed in new clusters, otherwise openstack clients will automatically try to use openstack designate
+	if strings.ToLower(opt.DNSType) == "none" {
+		if cluster.Spec.Networking.Topology == nil {
+			cluster.Spec.Networking.Topology = &api.TopologySpec{
+				DNS: api.DNSTypeNone,
+			}
+		} else {
+			cluster.Spec.Networking.Topology.DNS = api.DNSTypeNone
+		}
+	}
```

Comment:
I thought that we already set this in a generic way for all cloud providers:
https://github.com/kubernetes/kops/blob/e85051b709e57a07fea062495c32fde6c4bc5c5a/upup/pkg/fi/cloudup/new_cluster.go#L1304-L1311

Reply:
@hakman yes its done for new clusters. But like the comment says the code is executed too late for openstack. 

openstack needs that information already in row https://github.com/kubernetes/kops/blob/master/upup/pkg/fi/cloudup/new_cluster.go#L337

but its initialised in https://github.com/kubernetes/kops/blob/master/upup/pkg/fi/cloudup/new_cluster.go#L378

if that information is not place in row 337, it will automatically try to initialise openstack designate because cluster.UsesNoneDNS() returns still false. Openstack designate will not work if that is not installed to openstack setup (like in our case).

So we could have generic model IF we could move that before row 337, but its difficult because in that case we should move other parts as well